ods_ci/tests/Tests/100__deploy/120__upgrades/120__pre_upgrades.robot

```diff
@@ -96,16 +99,17 @@ Verify Model Can Be Deployed Via UI For Upgrade
     Open Model Serving Home Page
     Serve Model    project_name=${PRJ_TITLE}    model_name=${MODEL_NAME}    framework=openvino_ir    existing_data_connection=${TRUE}
     ...    data_connection_name=model-serving-connection    model_path=openvino-example-model
-    ${runtime_pod_name} =    Replace String Using Regexp    string=${RUNTIME_NAME}    pattern=\\s    replace_with=-
-    ${runtime_pod_name} =    Convert To Lower Case    ${runtime_pod_name}
     Run Keyword And Continue On Failure  Wait Until Keyword Succeeds
     ...  5 min  10 sec  Verify Openvino Deployment    runtime_name=${runtime_pod_name}
     Run Keyword And Continue On Failure  Wait Until Keyword Succeeds  5 min  10 sec  Verify Serving Service
     Verify Model Status    ${MODEL_NAME}    success
     Set Suite Variable    ${MODEL_CREATED}    True
     Run Keyword And Continue On Failure    Verify Model Inference    ${MODEL_NAME}    ${INFERENCE_INPUT_OPENVINO}    ${EXPECTED_INFERENCE_OUTPUT_OPENVINO}    token_auth=${FALSE}
     Remove File    openshift_ca.crt
-    [Teardown]   Dashboard Test Teardown
+    [Teardown]   Run Keywords    Dashboard Test Teardown
+    ...    AND
+    ...    Run Keyword If Test Failed    Get Events And Pod Logs    namespace=${PRJ_TITLE}
+    ...    label_selector=name=modelmesh-serving-${runtime_pod_name}
```

Comment:
This might not work if the test fails before entering execution, i.e. during the Setup. Perhaps it would be a good idea to initialize runtime_pod_name at a suite level and run a check against it before running this keyword?

Reply:
if the test fails at suite setup leve there won't be nothing to fetch logs from

Reply:
Yes, that's the problem, but it will also fail trying to access a variable that does not exist (IIRC the Teardown is always executed) - if we initialize the var with a known name (e.g. "no-pod-yet") we can have a small check within the teardown to make sure that the variable exists and it doesn't have that name, thus avoiding potential problems of no logs to fetch.

Reply:
what I mean to say is that if the test fails at setup, there won't be no pod to check

Reply:
done!

Reply:
That wasn't what I had in mind, you could check against the value of the variable and decide to not fetch logs from the pod(s) but rather only the events of the namespace. 
I'll approve for now but imho this could be improved.

Reply:
you're right sorry, I mixed it up a bit. Anyways, I don't think there are many benefits from that logic checking the `no-pod-yet` because at the end of the the test would be failed in any case if the pod didn't come up. no?